### DISCUSSION TITLE
Improvements for the finance window

### DISCUSSION THREAD

**Discussion by LC-Zorg** [OpenTTD/OpenTTD#9293] [Features]
I've been keeping these graphics for over two years. Perhaps some of you have been thinking about some financial enhancements in the game, meanwhile JGR has added the ability to borrow or repay any loan amount, so perhaps these graphics will be helpful in some way. I am far from saying that these solutions are the best. Perhaps even the help of these graphics will lie in the fact that they will show you which way not to go. ;) In any case, it is rather a starting point for what can be done and changed.
![New window Finances v1 1 3](https://user-images.githubusercontent.com/67662046/119268265-71c2be80-bbf2-11eb-9df2-b5ffce865d53.png)

Version with added parts of improved shares, taxes and detailed financial information. These would not be permanent elements, but only visible when such options are enabled. Hence the visible wider lines separating the individual parts.
![New finances window v4 1](https://user-images.githubusercontent.com/67662046/119268271-7ab39000-bbf2-11eb-9ef7-4f4f5cab19cf.png)

Version with icons that open additional windows that would show more detailed data (list icon) or general information (icon with the letter "i"). In most cases, icons would open existing lists simply with the appropriate sort selected (Revenue of Trains > open Trains list with sorting by profit last year). Instead of the icons, the list could be opened by clicking on the inscription, but I think this way could be more readable for newer players.
Shaded values ​​for shares, taxes and financial result are projected values. Perhaps good if taxes and shares would be calculated once a year. These, however, can be calculated even on an ongoing basis, then this form is unnecessary.
![New finances window v5 1](https://user-images.githubusercontent.com/67662046/119268275-8010da80-bbf2-11eb-95c4-daa14e7d2c71.png)

Small versions of the window with a loan bar added.
![New finances window mini v1 1](https://user-images.githubusercontent.com/67662046/119268280-84d58e80-bbf2-11eb-8c54-2e116cd002a7.png)

**Comment by benda** [comment]
I think this is a great start. 

For background about me, I work in finance and have a business degree.

Standard financial reporting consists of three separate documents - an income statement, a balance sheet, and a statement of cash flows.

The "financials" window in the game seems to be intended primarily as an income statement, with some balance sheet information at the bottom. Rather than try to incorrectly put this information into one screen, the game should have separate windows for an income statement and balance sheet. These are separate concepts entirely that shouldn't be grouped together. The loan functionality can be on the balance sheet window or a separate window entirely (and it would be nice to have more financing options than just a single loan).


The main idea of an income statement is to see all revenues followed by all expenses followed by net income (since roughly speaking revenue - expenses = net income). Net income is the most important number on an income statement, and this term isn't even found on the existing financials window. 


Income statement examples:
https://en.wikipedia.org/wiki/Income_statement
https://www.investopedia.com/terms/i/incomestatement.asp
https://www.zoho.com/books/guides/what-is-an-income-statement.html

Real world examples of these financial statements for a publicly traded railroad in the USA (CSX Corporation):
https://seekingalpha.com/symbol/CSX/income-statement
https://seekingalpha.com/symbol/CSX/balance-sheet
https://seekingalpha.com/symbol/CSX/cash-flow-statement

**Comment by nielsmh** [comment]
I think you need to separate "improvements for the GUI" from "changes to the economic model of the game".

The former can make the game more straightforward to play but doesn't change the rules. For example shortening the texts, separating and moving around things in the window.
All of that is relatively straight-forward. But please try to stick to the established design language of the GUI instead of inventing a new one.

The latter is changing the rules of the game. Things like variable interest rate, profit sharing with shareholders, taxes.
Changing the rules requires a lot of care to make sure things feel fair, don't get too much harder or easier, and the concepts are understandable. Also that it doesn't add excessive new computational/memory requirements to the game. There are many things with ownership and value of property, as well as appreciation and depreciation that isn't managed properly at the moment. I think many of your ideas adding new features to the economic model require ground work of this type to be feasible.

**Comment by benda** [comment]
I agree with nielsmh, I am talking about taking numbers the game already has and showing them appropriately. I see now that OP has a few lines for taxes, share income, etc., I was focused on just the GUI improvements, the first screenshot above.
